src/test/java/jp/kusumotolab/kgenprog/ga/mutation/selection/DefaultVariantSelectionTest.java

```diff
@@ -144,26 +143,69 @@ public void testOrderOfVariants() {
         .doesNotContainSequence(9L, 19L, 7L, 8L, 17L);
   }
 
+  /**
+   * BulidFailedの個体が選択されるかどうかをテストする.
+   */
+  @Test
+  public void testBuildFailed() {
+    final int variantSize = 12;
+    final DefaultVariantSelection variantSelection =
+        new DefaultVariantSelection(variantSize, random);
+    final List<Variant> current = new ArrayList<>();
+    final List<Variant> generated = new ArrayList<>();
+
+    // 2個に1個はbuildFailedの個体にする
+    setupLists(current, generated, 10,
+        e -> e % 2 == 0 ? EmptyTestResults.instance : new TestResults());
+
+    final List<Variant> selectedVariants = variantSelection.exec(current, generated);
+
+    // 12個選択するが，buildSuccessが10個しかないため10個しか返ってこない
+    assertThat(selectedVariants).hasSize(10)
+        .allMatch(Variant::isBuildSucceeded);
+  }
+
   /**
    * Variantを生成するメソッド.Fitnessのみを指定する.
    *
    * @param fitness Variantの持つFitness
    * @return variant 生成したVariant
    */
   private Variant createVariant(final Fitness fitness) {
-    final Variant variant = new Variant(0, 0, null, null, null, fitness, null, null);
-    return variant;
+    final TestResults testResults = new TestResults();
+    return createVariant(fitness, 0, testResults);
   }
 
   /**
-   * Variantを生成するメソッド.FitnessとIdを指定する.
+   * Variantを生成するメソッド.Fitness, Id, TestResultsを指定する.
    *
    * @param fitness Variantの持つFitness
    * @param id Variantに固有の値
+   * @param testResults Variantのテスト情報
    * @return variant 生成したVariant
    */
-  private Variant createVariant(final Fitness fitness, final int id) {
-    final Variant variant = new Variant(id, 0, null, null, null, fitness, null, null);
+  private Variant createVariant(final Fitness fitness, final int id,
+      final TestResults testResults) {
+    final Variant variant = new Variant(id, 0, null, null, testResults, fitness, null, null);
     return variant;
```

Comment:
`variant` を持たなくてもそのまま返せる
```suggestion
    return new Variant(id, 0, null, null, testResults, fitness, null, null);
```